src/visual.ts

```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
 
                 this.behavior.renderSelection(false);
             }
+
+            this.axes.attr("transform", SVGUtil.translate(this.margin.left, 0));
+            this.axisX.attr("transform", SVGUtil.translate(0, this.viewport.height - this.margin.bottom));
+            this.axisY.attr("transform", SVGUtil.translate(0, this.margin.top));
+
+            this.calculateAxes();
+            let xShow: boolean = this.data.settings.categoryAxis.show;
+            let yShow: boolean = this.data.settings.valueAxis.show;
+            if (xShow) {
```

Comment:
These statements below are very similar. Can we encapsulate them into a function?